### PR TITLE
add bootstrapping secrets page

### PR DESCRIPTION
**Pull request by enekofb**
Closes https://github.com/weaveworks/weave-gitops/pull/3476

**What changed?**
Docs added for bootstrapping secrets

**How did you validate the change?**

Screenshot 2023-03-29 at 18 02 28

### CODE REVIEW

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 1]
```diff
@@ -5,17 +5,274 @@ hide_title: true
...
+```
+
+More info about the CRD spec [here](./spec/v1alpha1/secretSync.mdx)
+
 ### Working with SecretSync
-// Step by step guide to how to set it up to acheive bootstrapping a secret from management cluster to a new cluster called 'dev'
 
+#### Before start
+
+1. You are using [Weave Gitops Enterprise version > XX](https://docs.gitops.weave.works/docs/releases/) with support for SecretsSync.
```
Comment (enekofb) [inline website/docs/secrets/bootstraping-secrets.mdx:76]: to review with @ahsayde minimum version available

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 2]
```diff
@@ -5,17 +5,274 @@ hide_title: true
...
+  targetNamespace: my-namespace
+```
+
+2. Check-in to your configuration repo within your managmenet cluster
+
+3. Create a PR, review and merge
+
+4. See the progress via UI // CLI // TODO does it have support?
+
+CLI
```
Comment (enekofb) [inline website/docs/secrets/bootstraping-secrets.mdx:106]: to review with @MostafaMegahid @ahsayde whether on this experience

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 3]
```diff
@@ -5,17 +5,274 @@ hide_title: true
...
+
+UI
+
+5. See the secret created in your leaf cluster
+
+CLI
```
Comment (enekofb) [inline website/docs/secrets/bootstraping-secrets.mdx:112]: same here as before

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 1 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx:76]: ticket to release it https://github.com/weaveworks/weave-gitops-enterprise/issues/2500

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 4]
```diff
@@ -5,17 +5,198 @@ hide_title: true
...
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: tf-runner
+    namespace: flux-system
+```
+
+### Bootstrap secrets
+
+//TODO how to distribute the module currently lives in wge
```
Comment (enekofb) [inline website/docs/secrets/bootstraping-secrets.mdx:165]: to par with @waleedhammam on how to distribute the modules

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 5]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"capi.weave.works/v1alpha1","kind":"SecretSync","metadata":{"annotations":{},"name":"my-dev-secret-syncer","namespace":"default"},"spec":{"clusterSelector":{"matchLabels":{"environment":"dev"}},"secretRef":{"name":"my-dev-secret"},"targetNamespace":"my-namespace"}}
+  creationTimestamp: "2023-03-28T16:44:37Z"
+  generation: 1
+  name: my-dev-secret-syncer
+  namespace: default
+  resourceVersion: "4980"
+  uid: f6670578-4403-46f8-a9b5-de8c80559c2b
+spec:
+  clusterSelector:
```
Comment (enekofb) [inline website/docs/secrets/bootstraping-secrets.mdx:122]: to review with @ahsayde and bootstrap controller v0.5.0

I cannot set the resource being reconciled

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 6]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+      environment: dev
+  secretRef:
+    name: my-dev-secret
+  targetNamespace: my-namespace
+```
+5. See the secret created in your leaf cluster
+
+CLI
+
+UI
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:133]: we dont show kubernetes secrets in the UI

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 7]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+  clusterSelector:
+    matchLabels:
+      environment: dev
+  secretRef:
+    name: my-dev-secret
+  targetNamespace: my-namespace
+```
+5. See the secret created in your leaf cluster
+
+CLI
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:131]: Just adding a comment here as a reminder to add the CLI example

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 8]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+
+UI
+
+## Using Terraform and `TF-controller`
+
+You could use terraform and tf-controller to provision the secrets. This scenario would be more suitable for an
+organisation that already leverages terraform for cluster management infrastructure and want to created integrated workflows.
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:138]:
```suggestion
organisation that already leverages terraform for cluster management infrastructure and wants to create integrated workflows.
```

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 9]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+    matchLabels:
+      environment: dev
+```
+
+More info about the CRD spec [here](./spec/v1alpha1/secretSync.mdx)
+
 ### Working with SecretSync
-// Step by step guide to how to set it up to acheive bootstrapping a secret from management cluster to a new cluster called 'dev'
 
+#### Before start
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:74]: We can use `Pre-requisites` instead of `Before start` to be consistent with other pages.

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 10]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+CLI
+
+UI
+
+## Using Terraform and `TF-controller`
+
+You could use terraform and tf-controller to provision the secrets. This scenario would be more suitable for an
+organisation that already leverages terraform for cluster management infrastructure and want to created integrated workflows.
+
+### Before start
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:140]: We can use `Pre-requisites` instead of `Before start` to be consistent with other pages.

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 11]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+```
+
+More info about the CRD spec [here](./spec/v1alpha1/secretSync.mdx)
+
 ### Working with SecretSync
-// Step by step guide to how to set it up to acheive bootstrapping a secret from management cluster to a new cluster called 'dev'
 
+#### Before start
+
+1. You are using [Weave Gitops Enterprise version > v0.19.0](https://docs.gitops.weave.works/docs/releases/#v0190)
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:76]: A bit of nitpicking from my side, in this section you use numbers, and in the other `before start` section you use bullet points. Would be nice to pick one and go with it everything.

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 12]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+│ The "for_each" map includes keys derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will
+│ identify the instances of this resource.
+│
+│ When working with unknown values in for_each, it's better to define the map keys statically in your configuration and place apply-time results only in the map values.
+│
+│ Alternatively, you could use the -target planning option to first apply only the resources that the for_each value depends on, and then apply a second time to fully converge.
+╵
+```
+
+- This is because the flux-system have to be applied before flux installation
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:211]:
```suggestion
- This is because the flux-system has to be applied before flux installation
```

**Review by serboctor — CHANGES_REQUESTED** [review]
Awesome and comprehensive guide. Just some small suggestions/simplificaitons here and there.

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 1 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: released and tested

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 13]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
-// State what we mean by bootstrapping secrets (Provide needed secrets for Profiles Helm Repo, Profiles, other applications needed in the cluster provisioning cluster)
-// State the 2 possible ways to do this (SecretSync, Terraform Modules)
+When accessing to protected resources there is a need for a client to authenticate before
+the access is granted and the resource is consumed. For the authentication, a client presents
+credentials that are either created manually or available through infrastructure a common scenario
+is a secrets store.
+
+Weave gitops allows you to provision the secret management infrastructure as part of [its capabilities](./setup-secrets-operator.mdx).
+However, in order to provision the secret infrastructure, as any other application that has secrets, we
+need to make the secret it needs to install it.
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: applied, thanks you 🙏

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 9 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx:73]: done ✔️

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 11 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx:75]: we should have now only one (given terraform is not yet in). happy to put bullet points if you consider makes more sense here.

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 12 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: not yet included this piece

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 8 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: terraform solution is not yet shipped so we could ignore

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 6 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: i guess it was more to show secretsync status from the UI

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 7 above)
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: done added yesterday

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 14]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+
 ### Working with SecretSync
-// Step by step guide to how to set it up to acheive bootstrapping a secret from management cluster to a new cluster called 'dev'
 
+#### Before start
+
+1. You are using [Weave Gitops Enterprise version > v0.19.0](https://docs.gitops.weave.works/docs/releases/#v0190)
+2. You are running cluster-bootstrap-controller > v0.5.0
+2. You have a secret that you want to sync from the management cluster.
+3. You have a set of GitopsClusters that you want to sync to.
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: added thanks you

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 15]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+Bootstrapping secrets will be not only present while provision your secrets managmenet solution but in
+any platform provisioning task that comes before you have avaialble before the secrets management solution
+is availabe. Another common example is provisioning platform capabilities itself via
+https://docs.gitops.weave.works/docs/cluster-management/getting-started/#profiles-and-clusters
+that are stored in private repositories. For example https://fluxcd.io/flux/guides/helmreleases/#helm-repository-authentication-with-credentials
+
+To address this problem, weave gitops provides the following alternatives:
+
+- SecretSync CRD: to leverage weave gitops for secrets bootstrapping.
+- SecretSync Terraform Module: to leverage terraform and t for secrets bootstrapping.
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: no longer tf docs

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 16]
```diff
@@ -5,17 +5,218 @@ hide_title: true
...
+is a secrets store.
+
+Weave gitops allows you to provision the secret management infrastructure as part of [its capabilities](./setup-secrets-operator.mdx).
+However, in order to provision the secret infrastructure, as any other application that has secrets, we
+need to make the secret it needs to install it.
+
+This type of initial secret is what we call it bootstrapping secret and this guide gives you
+with the different alternatives that weave gitops gives you to achieve the task.
+
+Bootstrapping secrets will be not only present while provision your secrets managmenet solution but in
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/secrets/bootstraping-secrets.mdx]: I reworded it yesterday before reading this comment. let me know whether the current version improves understanding or whether we should make it simpler.

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 17]
```diff
@@ -2,20 +2,151 @@
...
+the access is granted and the resource is consumed. For authentication, a client presents
+credentials that are either created manually or available through infrastructure. A common scenario
+is to have a secrets store.
+
+Weave Gitops allows you to provision the secret management infrastructure as part of [its capabilities](./setup-secrets-operator.mdx).
+However, in order to provision, as any other application that has secrets, we need to create the secret needed for installing it.
+This is known as a chicken-egg scenario that get addressed by providing an initial secret. This secret we call it
+bootstrapping secret.
+
+Bootstrapping secrets will be, not only present while provision your secrets management solution, but in
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:19]: I still find this whole paragrap a bit complex and hard to follow. If we are trying to tell the user that the `bootstrapping secrets` functionally is beneficial for other things as well, then we can say something like:
`Bootstrapping secrets comes in handy, not only while provisioning your secrets management solution, but also in any platform provisioning task where the existence of the secret is a prerequisite.  Another common example could be provisioning
platform capabilities via [profiles](../../../cluster-management/getting-started/#profiles-and-clusters)
that are stored in [private repositories](https://fluxcd.io/flux/guides/helmreleases/#helm-repository-authentication-with-credentials).`

**Review by serboctor — CHANGES_REQUESTED** [review]

File: website/docs/secrets/bootstraping-secrets.mdx (hunk 11 above)
Comment (serboctor) [reply website/docs/secrets/bootstraping-secrets.mdx]: I dont have a preference. Whatever makes more sense to you/is more consistent with the website.

File: website/docs/secrets/spec/index.mdx [hunk 18]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (serboctor) [inline website/docs/secrets/spec/index.mdx:1]: the whole spec directory is not showing when I run the docs locally. I think you need to add it to the sidebars file

**Review by serboctor — CHANGES_REQUESTED** [review]

File: website/docs/secrets/spec/index.mdx (hunk 18 above)
Comment (enekofb) [reply website/docs/secrets/spec/index.mdx:1]: done!

File: website/docs/secrets/bootstraping-secrets.mdx [hunk 19]
```diff
@@ -2,20 +2,151 @@
...
+Another common example could be provisioning platform capabilities via [profiles](../../../cluster-management/getting-started/#profiles-and-clusters)
+that are stored in [private repositories](https://fluxcd.io/flux/guides/helmreleases/#helm-repository-authentication-with-credentials).
+
+Weave Gitops provides [SecretSync](#secretsync) as a solution to managing your bootstrapping secrets.
+
+## SecretSync
+
+<AlphaWarning/>
+
+`SecretSync` is a [Kubernetes Customer Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
```
Comment (serboctor) [inline website/docs/secrets/bootstraping-secrets.mdx:30]:
```suggestion
`SecretSync` is a [Kubernetes Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
```

**Comment by enekofb** [comment]
@serboctor thanks so much for your thorough review ❤️
